### PR TITLE
JS API: prepare for compressed textures.

### PR DESCRIPTION
**Pull request by prideout**
This preps for adding compressed texture support to the JavaScript API.

Note that we need to allow clients to find out if compressed textures
are supported BEFORE the engine is created, so that they can start
downloading the correct set of texture files. To allow this, we now
expose a getSupportedFormats function that creates a throw-away canvas
in order to query extensions (creating a temporary canvas is a technique
that we also used in the old PNG decoder that we removed in 7b36ed4)

### CODE REVIEW

File: web/filament-js/jsbindings.cpp [hunk 1]
```diff
@@ -148,13 +148,23 @@ struct BufferDescriptor {
...
         this->pbd.reset(new driver::PixelBufferDescriptor(malloc(byteLength), byteLength,
                 fmt, dtype, [](void* buffer, size_t size, void* user) { free(buffer); }));
     }
+    // Note that embind allows overloading based on number of arguments, but not on types.
+    // It's fine to have two constructors but they can't both have the same number of arguments.
+    PixelBufferDescriptor(val arrdata, driver::CompressedPixelDataType cdtype, int imageSize,
+            bool compressed) {
+        auto byteLength = arrdata["byteLength"].as<uint32_t>();
+        assert(compressed == true);
+        assert(imageSize == byteLength || imageSize == byteLength / 6);
```
Comment (bejado) [inline web/filament-js/jsbindings.cpp]: Maybe a comment explaining where 6 comes from?

File: web/filament-js/utilities.js [hunk 2]
```diff
@@ -59,6 +59,24 @@ Filament.PixelBuffer = function(typedarray, format, datatype) {
...
+/// CompressedPixelBuffer ::function:: Constructs a [PixelBufferDescriptor] for compressed texture
+/// data by copying a typed array into the WASM heap.
+/// typedarray ::argument:: Data to consume (e.g. Uint8Array, Uint16Array, Float32Array)
+/// cdatatype ::argument:: [CompressedPixelDataType]
+/// ::retval:: [PixelBufferDescriptor]
+Filament.CompressedPixelBuffer = function(typedarray, cdatatype) {
+    console.assert(typedarray.buffer instanceof ArrayBuffer);
+    console.assert(typedarray.byteLength > 0);
+    if (Filament.HEAPU32.buffer == typedarray.buffer) {
+      typedarray = new Uint8Array(typedarray);
```
Comment (bejado) [inline web/filament-js/utilities.js]: The rest of the file uses 4 space indent

File: web/filament-js/jsbindings.cpp (hunk 1 above)
Comment (prideout) [reply web/filament-js/jsbindings.cpp]: Good catch, done.

File: web/filament-js/utilities.js (hunk 2 above)
Comment (prideout) [reply web/filament-js/utilities.js]: Good catch, fixed.

**Pull request merged**